src/fides/api/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -2179,15 +2180,21 @@ def get_test_privacy_request_results(
         db, results, dataset_key, privacy_request.policy_id  # type: ignore[arg-type]
     )
 
-    return {
-        "privacy_request_id": privacy_request.id,
-        "status": privacy_request.status,
-        "results": (
-            filtered_results
-            if CONFIG.security.dsr_testing_tools_enabled
-            else "DSR testing tools are not enabled, results will not be shown."
-        ),
-    }
+    with logger.contextualize(
+        privacy_request_id=privacy_request.id,
+        privacy_request_source=privacy_request.source.value,
+    ):
```

Comment:
This is another key change. There are instances where we can't use the `@log_context` decorator because we're not passing in any params of type `PrivacyRequest` that we can automatically extract the ID and source from. In those cases we use a manual `with logger.contextualize` context